### ISSUE TITLE
BREAKING: Refactor workflow/activity options

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#820]
Currently the `[WorkflowInterface]` and `[ActivityInterface]` attributes support only custom `TaskList` properties.  We should add `Namespace` to this as well so that creating stubs for an interface will work without needing the caller to explicitly know the target namespace.

**Comment by jefflill** [comment]
I thing we need to do a somewhat bigger refactor here:

* First, it's a bit weird that `TaskList` is specified in `ActivityOptions`, `WorkflowOptions`, and `ChildWorkflowOptions` but `Domain` is not.  Most client methods that accept options also accept an optional `domain` argument.  We followed the Cadence Java client here, but this really doesn't make sense to me.  I think we should probably make domains be consistent.

* Looking at the stub manager code, we're ignoring task lists  for workflows and activities.  This would be a good time to address this as well.

So, we're going to go ahead and:

- [ ] Add `Domain/Namespace` to `ActivityOptions`, `WorkflowOptions`, and `ChildWorkflowOptions`.

- [ ] **BREAKING:** Remove all of the optional `domain/@namespace` method parameters.

- [ ] Add `Domain/Namespace` to `ActivityInterfaceAttribute` and `WorkflowInterfaceAttribute`.

- [ ] `WorkflowMethodAttribute` and `ActivityMethodAttribute` both include `TaskList` properties.  These should also add `Domain/Namespace` properties to be consistent.

- [ ] Fix `StubManager` to handle tasklist correctly as well as adding domain/namespace interface attribute support.

Here's how we'll implement precedence for these:

1. **highest:** Values in options have the highest precedence.
2.  Values specified in method attributes are next.
3. Followed by values specified in interface attributes
4. Finally (for domain/namespace only), we'd use the default domain specified by the client settings.

**Comment by jefflill** [comment]
This turned in to a very deep 2-day rabbit hole.

The problem was that although I started to centralize how workflow and activity options are processed via the `WorkflowOptions.Normalize()`, `ChildWorkflowOptions.Normalize()`, and `ActivityOptions.Normalize()` methods a few months back, there was still a substantial amount of older normalization code sprinkled throughout the the library.  Much of this code was incomplete and inconsistent.  So I upgraded the normalization methods above to handle all cases and did some major surgery to ensure that these methods were called everywhere (replacing any old one-off code).

**BREAKING:** I also renamed some of the options property and also added the `CadenceSettings.DefaultTaskList` and `TemporalSettings.DefaultTaskList` properties.

All of the Cadence unit tests are passing again.  I'll add additional tests for #669 to actually verify that these settings behave correctly.
